src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -24,30 +24,34 @@ private class StateSet
 
             // analyzer version this state belong to
             private readonly VersionStamp _analyzerVersion;
-            private readonly AnalyzerTypeData _analyzerTypeData;
+            private readonly AnalyzerStateNames _analyzerStateNames;
 
             private readonly ConcurrentDictionary<DocumentId, ActiveFileState> _activeFileStates;
             private readonly ConcurrentDictionary<ProjectId, ProjectState> _projectStates;
 
+            // whether this analyzer has compilation end analysis or not
+            // -1 not set, 0 no, 1 yes.
+            private volatile int _compilationEndAnalyzer = -1;
```

Comment:
Consider nullable bool.

Reply:
used int so that I can make it volatile.